packages/block-editor/src/components/list-view/use-block-selection.js

```diff
@@ -114,6 +117,16 @@ export default function useBlockSelection() {
 			// the total number of blocks deselected is greater than one.
 			const updatedSelectedBlocks = getSelectedBlockClientIds();
 
+			// If the selection is greater than 1 and the Home or End keys
+			// were used to generate the selection, then skip announcing the
+			// deselected blocks.
+			if (
+				( event.keyCode === HOME || event.keyCode === END ) &&
+				updatedSelectedBlocks.length > 1
+			) {
+				return;
+			}
+
```

Comment:
In testing on both Windows with NVDA+Firefox and macOS with VoiceOver+Firefox, this change doesn't seem to make any difference. The last selected block of the multi-selection is always announced, whether the multi-selection happens with Home/End keys or just Shift+Arrow Up/Down. And if for @alexstine the change seemed to make things worse, I think it's safe to remove it altogether.

Reply:
This change is good when selecting with Arrows but when using Home/End, it should only announce the number of selected/deselected. Why? Let's look at this practically.

- I press Shift+Down Arrow to select a block.
- I hear the total number of blocks selected plus the block I just selected.
- If I press End, it will select say 5 blocks but then I only hear the last block get announced.

See how this can get confusing?

When having these Home/End selections, it should just announce the total as in reality you selected multiple blocks at once vs. one block at a time.

Reply:
I see what you mean. When pressing End, the last block in the selection gets focused; that must be why it's being announced. I wonder if we can instead keep focus on the first block in the selection - and if there won't be any negative side-effects to that 😅

Reply:
Thanks for taking a closer look here! I was wondering if it's possible for us to conditionally add `aria-hidden` set to `true` for just the case of having switched focus while holding shift plus Home / End, to prevent the block from being announced. I might have a play.

Reply:
I think focus should always follow the selection. Otherwise when you press up or down again it becomes disorienting.

Looking at how Finder works, it's similar to what Alex describes (though you have to use shift+option+arrow key instead of home or end to select to the start or end). It'll only announce '6 rows selected' and it won't announce the file that is focused (even though focus does move).

Is there any way to suppress the innate announcement that screenreaders make when focus moves?

Reply:
I was also wondering if there's any advantage in using roles like `aria-multiselectable`:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-multiselectable

It didn't seem to do anything when I tested adding it, but maybe it needs to be used in combination with `aria-selected`.

The docs seem to imply that's ok with a `tree`, but don't say if that extends to a `treegrid`.